File: api/routers/pipeline.py
"""Pipeline API router — v0.5.1 P1 endpoint for pipeline command preview."""
from fastapi import APIRouter, Depends

from api.dependencies import get_current_user
from api.schemas import PipelineRunRequest, PipelineRunResponse
from services.pipeline_service import preview_pipeline_run

router = APIRouter(dependencies=[Depends(get_current_user)])


@router.post("/pipeline/run", response_model=PipelineRunResponse)
def pipeline_run(body: PipelineRunRequest):
    result = preview_pipeline_run(body.pipeline_type)
    return PipelineRunResponse(
        command=result.get("command", ""),
        pipeline_type=result.get("pipeline_type", body.pipeline_type),
        estimated_duration=result.get("estimated_duration", ""),
        required_env_vars=result.get("required_env_vars", []),
        warnings=result.get("warnings", []),
        description=result.get("description", ""),
    )

File: internal/api/dto/pipeline.go
package dto

// PipelineRunRequest is the request body for POST /api/v1/pipeline/run.
type PipelineRunRequest struct {
	// Config is the pipeline configuration name (e.g. "ingest_raw", "full").
	Config string `json:"config"`
}

// PipelineRunResponse is the response body for POST /api/v1/pipeline/run.
type PipelineRunResponse struct {
	RunID  string `json:"run_id"`
	Status string `json:"status"`
}

// PipelinePreview is the preview response for a pipeline run.
type PipelinePreview struct {
	Command           string   `json:"command"`
	PipelineType      string   `json:"pipeline_type"`
	EstimatedDuration string   `json:"estimated_duration"`
	RequiredEnvVars   []string `json:"required_env_vars"`
	Warnings          []string `json:"warnings"`
	Description       string   `json:"description"`
}

// PipelineInfo describes an available pipeline type.
type PipelineInfo struct {
	Type        string `json:"type"`
	Description string `json:"description"`
}
